aws-crt/lib/aws-crt/io.rb

```diff
@@ -0,0 +1,23 @@
+module Aws::Crt
+  module IO
+    # A collection of event-loops.
+    # An event-loop is a thread for doing async work, such as I/O.
+    # Classes that need to do async work will ask the EventLoopGroup for an event-loop to use.
+    class EventLoopGroup
+      def initialize(max_threads = nil)
+        unless max_threads.nil? || (max_threads.is_a?(Integer) && max_threads.positive?)
+          raise ArgumentError, 'max_threads must be nil or positive Integer'
+        end
+
+        max_threads = 0 if max_threads.nil?
+
+        @native = Aws::Crt.call { Aws::Crt::Native.event_loop_group_new(max_threads) }
+      end
+
+      # this function is going away in the near future, just ignore it
+      def destroy
+        Aws::Crt::Native.event_loop_group_destroy(@native)
```

Comment:
Will this be moved to a finializer?  If so - read this https://www.mikeperham.com/2010/02/24/the-trouble-with-ruby-finalizers/  - its easy to write a ruby finializer in the wrong way and ensure that the object will never get cleaned up by GC (because the finalizer has a reference to the object).

Reply:
Yeah, this will become a finalizer. I just wasn't ready to mess with finalizers yet. I will read the warnings before writing any

Reply:
As we develop things further, we'll add checks to ensure we're not leaking any memory. So that will catch any mis-written finalizers.